### PR TITLE
fix: correct misleading comments and docs in BinaryJointAction

### LINKED ISSUE #4670: [Bug Report] BinaryJointAction.process_actions: bool-action convention mismatch / confusing comment (True treated as open, False as close)
If you are submitting a bug report, please fill in the following details and use the tag [bug].

### Describe the bug
With torch.bool, False == 0 evaluates to True, so False is mapped to close_command and True is mapped to open_command (because torch.where(binary_mask, close, open) is used).

This behavior appears to contradict the inline comment (true: close, false: open) and can also be interpreted as inconsistent with a common expectation that True means “activate/close”. Even if the intended convention is “Open=1/True, Close=0/False”, the current comment is misleading and likely to cause user confusion / integration bugs.

Code reference [Binary Joint Action](https://github.com/isaac-sim/IsaacLab/blob/e10c302cd247eb98201067952fcd38979074fe3d/source/isaaclab/isaaclab/envs/mdp/actions/binary_joint_actions.py#L135-L137)

### Steps to reproduce

```
import torch

# Dummy open/close commands for 2 joints
open_command  = torch.tensor([1.0, 1.0])
close_command = torch.tensor([0.0, 0.0])

def process_bool(actions_bool: torch.Tensor):
    # current repo logic
    binary_mask = actions_bool == 0   # False -> True, True -> False
    processed = torch.where(binary_mask, close_command, open_command)
    return binary_mask, processed

actions = torch.tensor([[True], [False]], dtype=torch.bool)
mask, out = process_bool(actions)

print("actions:\n", actions.squeeze(-1))
print("binary_mask (True => close):\n", mask.squeeze(-1))
print("processed:\n", out)
```

Observed output:

actions=True -> binary_mask=False -> selects open_command

actions=False -> binary_mask=True -> selects close_command

So for torch.bool inputs: True => open, False => close.

### System Info

Describe the characteristic of your environment:

- Commit: [e.g. 8f3b9ca]
- Isaac Sim Version: [e.g. 5.0, this can be obtained by `cat ${ISAACSIM_PATH}/VERSION`]
- OS: [e.g. Ubuntu 22.04]
- GPU: [e.g. RTX 5090]
- CUDA: [e.g. 12.8]
- GPU Driver: [e.g. 553.05, this can be seen by using `nvidia-smi` command.]

### Additional context

Potential fix options:

If intended behavior is Open=True, Close=False:

Keep logic but fix the comment to match reality, e.g.:

# binary_mask: True -> close, False -> open

or # actions: True -> open, False -> close

If intended behavior is Close=True, Open=False:

Change the bool branch to something like:

binary_mask = actions (if True means close)

or binary_mask = actions != 0

and ensure comments + docs match.

Right now the mismatch between code and comment makes it easy for downstream code to pass boolean actions with the wrong meaning.
### Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have checked that the issue is not in running Isaac Sim itself and is related to the repo

### Acceptance Criteria

Add the criteria for which this task is considered **done**. If not known at issue creation time, you can add this once the issue is assigned.

- [ ] Criteria 1
- [ ] Criteria 2

### PR DESCRIPTION
**Pull request by alltheseas**
## Description

Clarify the established binary-action convention without changing behavior:

- `True` and non-negative floating-point actions select the configured open command.
- `False` and negative floating-point actions select the configured close command.
- Rename the internal selection mask to `close_mask` so its meaning cannot be confused with the input action.
- Apply the same clarification to both joint-based and surface-gripper binary actions, which share this contract.
- Correct the documented zero boundary: zero opens because the close condition is strictly `< 0`.

The runtime convention is intentionally preserved. Reversing the boolean mapping would break existing policies and downstream integrations that follow the documented `True`/positive = open contract.

The earlier `resolve_matching_names` change is no longer part of this PR. Both affected string helpers were fixed comprehensively by #5546.

Fixes #4670

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Documentation update

## Release backport

- [x]  Backport this pull request to the active `release/3.0.0` branch after it merges into `develop`

This clarification applies to the same implementation on `release/3.0.0`; please backport it so the supported 3.0 documentation and source comments do not retain the ambiguous convention or incorrect zero boundary.

## Validation

- Exercised both action terms with boolean and floating-point inputs, including `True`, `False`, positive, zero, and negative values.
- `uv run isaaclab -f`
- `uv run python tools/changelog/cli.py check develop`
- `uv run --isolated --extra dev -- make -C docs current-docs`

## Checklist

- [x] I have read and understood the contribution guidelines.
- [x] I have run the pre-commit checks.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added a changelog fragment under `source/isaaclab/changelog.d/`.
- [x] The contributor is listed in `CONTRIBUTORS.md`.

### CODE REVIEW

**Comment by ooctipus** [comment]
HI @alltheseas  thanks for your PR, though I am still a bit unclear what is the confusion? the doc does reads correct to me, binary mask evaluate to True if its 0 or less than 0, and False if its greater than 0.

then all true(<=0) is map to close, and all false(>0) is mapped to open

I also see there are update to preserve order that I am not sure the newer version is correct. @greptile for comment.

**Review by ooctipus — APPROVED** [review]
Thanks for making the doc clearer :D

**Comment by alltheseas** [comment]
Thanks - please treat my contributions with a risk of hallucinations. 

To be clear, the issue/confusion was described in https://github.com/isaac-sim/IsaacLab/issues/4670, which is why I submitted the smol PR. 

🙏

**Comment by purewater0901** [comment]
@alltheseas Thank you for the PR and for addressing the issue I posted! Is this going to be merged soon?

**Comment by StafaH** [comment]
run-ci

**Review by StafaH — APPROVED** [review]

**Comment by StafaH** [comment]
run-ci
